source/DataWriter.h

```diff
@@ -47,45 +49,71 @@ class DataWriter {
 
 	// Save the contents to a file.
 	void SaveToPath(const std::string &path);
+	// Save the contents to a stream.
+	void SaveToStream(std::ostream &stream);
+	// Pass the contents to a function.
+	void SaveToFunction(const std::function<void(std::string)> func);
```

Comment:
I would replace all of these with a function that returns the whole content as a string.

Reply:
Should I remove 'SaveToPath' as well?

Reply:
yup